### PR TITLE
Move Nimble out of Nim compiler wrapper

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Nimble doesn't need to be bundled with the Nim compiler and shouldn't be tied to a specific Nim compiler.

Package the alternate package tool atlas.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/nim-packages/nimble/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    repo = "nimble";
+    rev = "v${final.version}";
+    hash = "sha256-8b5yKvEl7c7wA/8cpdaN2CSvawQJzuRce6mULj3z/mI=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  doCheck = false; # it works on their machine
+
+  wrapperArgs = [ "--suffix PATH : ${lib.makeBinPath [ nim ]}" ];
```
Comment (FRidh) [inline pkgs/development/nim-packages/nimble/default.nix:18]: not needed since args are already in postInstall

**Comment by Kranzes** [comment]
Please rename commits to comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

**Comment by ehmry** [comment]
> Please rename commits to comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Please be more specific so I don't have to fix the messages twice.

**Comment by Kranzes** [comment]
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

**Comment by lf-** [comment]
Hi, this broke master and causes eval errors on ofborg: https://gist.github.com/GrahamcOfBorg/7a33204709bf948180670867a9e22e02. I will push a new PR to fix it.

**Pull request merged**